CHANGELOG.datadog.md

```diff
@@ -0,0 +1,4 @@
+## 2024-10-18
```

Comment:
I'm unsure what this version should be. Could it be `3.3.6.1` so that we stick with the upstream `3.3.6` version?

Reply:
I think this is fine unless/until a problem with just adding `.1` at the end is discovered.

Reply:
I'd keep the same version if all we did was remove unsupported versions of Ruby.
We can use `3.3.6.x` for bug fixes if needed.